embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/comms/delivery/RateLimitHandler.kt

```diff
@@ -0,0 +1,26 @@
+package io.embrace.android.embracesdk.comms.delivery
+
+import io.embrace.android.embracesdk.comms.api.EmbraceApiService.Companion.Endpoint
+
+internal interface RateLimitHandler {
```

Comment:
I need to think about this some more, but the `RateLimitHandler` is depended on by 3 different components, and 1 of them is set after initialization. I wonder if there's a better way to shape this responsibilities of this component so that it needs to be invoked directly by fewer components?

Reply:
I moved the logic to the Endpoint class and now it looks much better. Maybe the scheduleRetry() method in the Endpoint class is not the best option, but in the future we can do a refactor to unify it in the PendingApiCallsSender without overloading that class too much.